### PR TITLE
:seedling: add migrationwave API test

### PR DESCRIPTION
**Pull request by khareyash05**
Added the migration wave API test

### CODE REVIEW

File: binding/migrationwave.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// Get a MigrationWave by ID.
+func (h *MigrationWave) Get(id uint) (r *api.MigrationWave, err error) {
+	r = &api.MigrationWave{}
+	path := Path(api.MigrationWaveRoot).Inject(Params{api.ID: id})
+	err = h.client.Get(path, r)
+	return
+}
+
+//
+// List Reviews.
```
Comment (jortel) [inline binding/migrationwave.go:30]: List MigrationWaves

File: test/api/migrationwave/api_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	// List MigrationWaves.
+	got, err := MigrationWave.List()
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	// Compare contents of migration waves.
+	for _, createdMigrationWave := range createdMigrationWaves {
+		found := false
+		for _, retrievedReview := range got {
```
Comment (jortel) [inline test/api/migrationwave/api_test.go:122]: retrievedWave

File: test/api/migrationwave/api_test.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+			if assert.FlatEqual(createdMigrationWave.ID, retrievedReview.ID) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Expected review not found in the list: %v", createdMigrationWave)
```
Comment (jortel) [inline test/api/migrationwave/api_test.go:129]: Expected wave

**Comment by khareyash05** [comment]
Oh missed that when using the same elements as Reviews. Will Update all. Thank you for the review.

**Comment by aufi** [comment]
Related to https://github.com/konveyor/operator/issues/220

File: test/api/migrationwave/api_test.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+		}
+
+		for _, stakeholderGroup := range createdMigrationWave.StakeholderGroups {
+			assert.Must(t, StakeholderGroup.Delete(stakeholderGroup.ID))
+		}
+		assert.Must(t, MigrationWave.Delete(createdMigrationWave.ID))
+	}
+}
+
+func AssertEqualMigrationWaves(t *testing.T, got *api.MigrationWave, expected api.MigrationWave) {
```
Comment (aufi) [inline test/api/migrationwave/api_test.go:150]: Add checks for associated resources (Applications, Stakeholders, StakeholderGroups) to check all attributes sent via API.

File: test/api/migrationwave/api_test.go [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+		}
+
+		for i, stakeholderGroup := range r.StakeholderGroups {
+			expectedStakeholderGroup := api.StakeholderGroup{
+				Name:        stakeholderGroup.Name + strconv.Itoa(i),
+				Description: "Sample Stakeholder Group",
+			}
+			assert.Must(t, StakeholderGroup.Create(&expectedStakeholderGroup))
+		}
+		assert.Must(t, MigrationWave.Create(&r))
```
Comment (aufi) [inline test/api/migrationwave/api_test.go:109]: Consider order of MigrationWave related objects deletion - try avoid deleting applications associated with an existing migrationwave, instead delete resources in an opposite order than in its creation, so 1. migration wave, then others.

**Review by aufi — COMMENTED** [review]
Overall looks good to me, added few comments. Thanks!

File: test/api/migrationwave/api_test.go (hunk 5 above)
Comment (khareyash05) [reply test/api/migrationwave/api_test.go:109]: Will Keep that in mind.

**Comment by aufi** [comment]
And update API test matrix (as in previous PRs)..

File: test/api/migrationwave/api_test.go [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+			}
+			assert.Must(t, Application.Create(&expectedApp))
+		}
+
+		for _, stakeholder := range r.Stakeholders {
+			expectedStakeholder := api.Stakeholder{
+				Name:  stakeholder.Name,
+				Email: "sample@example.com",
+			}
+			assert.Must(t, Stakeholder.Create(&expectedStakeholder))
```
[the remark this answers is not in the document]
Comment (khareyash05) [reply test/api/migrationwave/api_test.go:27]: Will double check on my local env

[the remark this answers is not in the document]
Comment (khareyash05) [reply test/api/migrationwave/api_test.go:27]: Ya that seems an issue.Noticed how. Thanks for the check will update them

**Comment by aufi** [comment]
Thanks for update, but I'm afraid I'm not OK with the latest change. Plus one on passing test and not leaving orphaned database records :+1: 

But not creating dependencies before creating migration wave doesn't looks right to me. The side-effect that the Create action on Migration wave creates dependencies from its Refs is something I don't see as correct (not saying it is a bug, just an edge case), not sure how my colleagues feel about this, but for my approval, please create (and delete) dependent objects for the migration wave manually. (that should match to typical use-case in UI where user creates migration wave first and then adds applications to the migration wave)

File: test/api/migrationwave/api_test.go [hunk 7]
```diff
@@ -11,29 +10,6 @@ import (
 
 func TestMigrationWaveCRUD(t *testing.T) {
 	for _, r := range Samples {
-		for _, app := range r.Applications {
-			expectedApp := api.Application{
-				Name:        app.Name,
```
Comment (aufi) [inline test/api/migrationwave/api_test.go]: Just add ID here before the Name

**Review by aufi — APPROVED** [review]
I have a small problem with creating dependencies in API call together with the MigratioWave, but since API allows it, not a blocker. LGTM.

**Comment by khareyash05** [comment]
Let's discuss this over call. Might have missed something.

**Pull request merged**